### PR TITLE
Mass shorten links with csv file upload

### LINKED ISSUE #58: add CSV to shortened links feature
This allows for the bulk shortnening of links from a CSV file.

### PR DESCRIPTION
**Pull request by BreakTos**
Demo of csv upload feature

https://youtu.be/xOzT0hMn208

For testing purposes here is a csv file with links column and a few urls in it
[SampleCSVFile_2kb.csv](https://github.com/AmoabaKelvin/ishortn.ink/files/14817473/SampleCSVFile_2kb.csv)

Optional aliases column remains to be implemented. closes #58

### CODE REVIEW

File: src/components/forms/dashboard/links/link-edit-form.tsx [hunk 1]
```diff
@@ -110,10 +167,19 @@ const LinkEditForm = () => {
 
       <section className="grid grid-cols-1 gap-5 mt-6 md:grid-cols-11">
         <div className="flex flex-col col-span-5 gap-4">
+
           <div className="flex flex-col gap-1">
-            <h1 className="text-2xl">Create your Link</h1>
+            <div className="options" style={{display:"flex",flexDirection:"row",justifyContent:"space-between"}}>
```
Comment (AmoabaKelvin) [inline src/components/forms/dashboard/links/link-edit-form.tsx]: can we replace these styles with tailwindcss classnames
```
<div className="flex justify-between options">
    ...
</div>
```

File: src/components/forms/dashboard/links/link-edit-form.tsx [hunk 2]
```diff
@@ -110,10 +167,19 @@ const LinkEditForm = () => {
...
+            <h1 className="text-2xl">Create your Link </h1>
+            <h1 className="text-2xl">Or</h1>
+            <button className="border border-white-900 border-2 px-3 py-2" onClick={()=>{
+              document.getElementById('inputCSV')?.click();
+            }}> Upload CSV</button>
```
Comment (AmoabaKelvin) [inline src/components/forms/dashboard/links/link-edit-form.tsx]: Let's use the `Button` component from the shadcn library we are using so we keep things consistent

**Review by AmoabaKelvin — CHANGES_REQUESTED** [review]
We can export this to it's own component rather than having everything in the link edit form. The new component will be the button that will be placed on the sidebar and every functionality related to the upload and shortening will be kept in there.

Also, if you can get `prettier` to format the codes, it will be awesome. Please let me know if there is anything else.

**Comment by BreakTos** [comment]
image

There we have it, Used prettier and made components to modulate the functionality

File: src/components/forms/dashboard/links/csv-shorten.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+          document.getElementById("inputCSV")?.click();
+        }}
+      >
+        {" "}
+        Upload CSV
+      </Button>
+      <input
+        type="file"
+        id="inputCSV"
+        style={{ display: "none" }}
```
Comment (AmoabaKelvin) [inline src/components/forms/dashboard/links/csv-shorten.tsx]: you can use the hidden className from tailwindcss here.

```html
<input
    type="file"
    id="inputCSV"
	className="hidden"
    onChange={handleCSV}
></input>
```

File: src/components/forms/dashboard/links/csv-shorten.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import { quickLinkShorten } from "@/actions/link-actions";
+import { Button } from "@/components/ui/button";
+
+import { useTransition } from "react";
+
+const CsvShortenForm = () => {
+  const [loading, startTransition] = useTransition();
+  interface CSVRow {
+    links: string;
+  }
```
Comment (AmoabaKelvin) [inline src/components/forms/dashboard/links/csv-shorten.tsx]: We should move the interface outside of the function so it's definition comes before the `CsvShortenForm`

File: src/components/forms/dashboard/links/csv-shorten.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+  function isURL(str: string): boolean {
+    const urlRegex = /^(?:https?|ftp):\/\/(?:\w+\.?)+/i;
+    return urlRegex.test(str);
+  }
+  const handleCSV = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file) {
+      console.log(file.type);
+      if (file.type === "text/csv") {
```
Comment (AmoabaKelvin) [inline src/components/forms/dashboard/links/csv-shorten.tsx:21]: Can we show an error toast if the filetype is not a CSV?

`toast.error("Please upload a CSV file")`

File: src/components/forms/dashboard/links/link-edit-form.tsx
Comment (AmoabaKelvin) [inline src/components/forms/dashboard/links/link-edit-form.tsx:1]: Let's remove the commented out codes for the CSV functionality from this file.

**Review by AmoabaKelvin — CHANGES_REQUESTED** [review]

**Comment by BreakTos** [comment]
Completed the above tasks, No idea why prettier check is failing for those 2 documents. I formatted the code with Prettier extension before committing.

**Pull request closed without merging**